src/host/init.cpp

```diff
@@ -30,38 +28,35 @@ void InitSideBySide(_Out_writes_(ScratchBufferSize) PWSTR ScratchBuffer, __range
     // make references to DLLs in the system that are in the SxS cache (ex. a 3rd party IME is loaded and asks for
     // comctl32.dll. The load will fail if SxS wasn't initialized.) This was bug# WIN7:681280.
 
-    // We look at the first few chars without being careful about a terminal nul, so init them.

```

Comment:
we have lost the invariant that the scratch buffer is cleared on failure. is that okay?

if you could write a bit about this change, it would be easier to accept! :)

Reply:
I checked the only caller of this function and the name of the argument is correct: It really is just a "scratch(pad) buffer" so that the callee doesn't need to allocate on the heap. It can be left uncleared on return. Additionally the caller calls this function last, right before returning and as such there's no one there to read from it anymore.